### DISCUSSION TITLE
Wizard scene with inline button

### DISCUSSION THREAD

**Discussion by weihan1394** [telegraf/telegraf#1367] [Q&A] [answered]
Hi, 

I am trying to use an inline-keyboard with WizardScene. How can i read the value from the callbackbutton? I believe I will have to add action before calling the scene. But is there any other way that I can retrieve the value? and at the same time clearing the inlineKeyboard after selecting?

```
let item;
const itemSelector = new Composer();
itemSelector.action('1.1', (ctx) => {
  item = '1.1';
  return ctx.wizard.steps[ctx.wizard.cursor - 1](ctx);
});
itemSelector.action('1.2', (ctx) => {
  item = '1.2';
  return ctx.wizard.steps[ctx.wizard.cursor - 1](ctx);
});
itemSelector.action('2.1', (ctx) => {
  item = '2.1';
  return ctx.wizard.steps[ctx.wizard.cursor - 1](ctx);
});
itemSelector.action('2.2', (ctx) => {
  item = '2.2';
  return ctx.wizard.steps[ctx.wizard.cursor - 1](ctx);
});

const responseScene = new WizardScene(
  'response',
  (ctx) => {
    ctx.reply('1', Markup.removeKeyboard().extra());
    return ctx.wizard.next();
  },
  (ctx) => {
    ctx.reply(
      'Which EDI you want to check leh? 😏',
      Markup.inlineKeyboard([
        [
          Markup.callbackButton('1.1', '1.1'),
          Markup.callbackButton('1.2', '1.2'),
          Markup.callbackButton('2.1', '2.1'),
        ],
        [
          Markup.callbackButton('2.2', '2.2'),
          Markup.callbackButton('2.3', '2.3'),
          Markup.callbackButton('3.1', '3.1'),
        ],
      ]).extra(),
    );
    ctx.wizard.state.response = {};
    return ctx.wizard.next();
  },
  itemSelector,
  (ctx) => {
    ctx.reply('what is the key! 🤭');
    return ctx.wizard.next();
  },
);

```

**Comment by weihan1394** [comment, marked as answer]
I was going through the example. But I don't understand why we need to include this. It the callback will only works if I click the button twice.
![image](https://user-images.githubusercontent.com/7189650/108022304-46608e80-705b-11eb-8d57-a0568f90dfa7.png)

Thankful if anyone can drop an advice.

**Reply by weihan1394** [reply]
I manage to retrieve the callback data with `ctx.callbackQuery.data`.

Closing the question.

**Reply by wojpawlik** [reply]
Try upgrading to `telegraf@4.0.3`

**Reply by weihan1394** [reply]
Actually why does the stephandler added this line below. What does it do? 

 stephandler.use(ctx)....

**Reply by weihan1394** [reply]
> Try upgrading to `telegraf@4.0.3`

After I upgrade to telegraf@4.0.3 I could not create a base scene. It keeps prompt me that Scenes.BaseScene is not a constructor.

I am currently developing in javascript and not typescript.

```
const currentScene = new Scenes.BaseScene('current');
```


Error:
```
(node:22508) UnhandledPromiseRejectionWarning: TypeError: Scenes.BaseScene is not a constructor
```

**Reply by alexey-sh** [reply]
So in case of two possible input source(user text and inline button) we have to setup two different callbacks. Does it look like a good pattern?
